src/main/java/io/github/apace100/apoli/power/ActionOnDeathPower.java

```diff
@@ -0,0 +1,48 @@
+package io.github.apace100.apoli.power;
+
+import io.github.apace100.apoli.Apoli;
+import io.github.apace100.apoli.data.ApoliDataTypes;
+import io.github.apace100.apoli.power.factory.PowerFactory;
+import io.github.apace100.calio.data.SerializableData;
+import net.minecraft.entity.Entity;
+import net.minecraft.entity.LivingEntity;
+import net.minecraft.entity.damage.DamageSource;
+import net.minecraft.util.Pair;
+
+import java.util.function.Consumer;
+import java.util.function.Predicate;
+
+public class ActionOnDeathPower extends Power {
+
+    private final Predicate<Pair<DamageSource, Float>> damageCondition;
+    private final Predicate<Pair<Entity, Entity>> bientityCondition;
+    private final Consumer<Pair<Entity, Entity>> bientityAction;
+
+    public ActionOnDeathPower(PowerType<?> type, LivingEntity entity, Predicate<Pair<DamageSource, Float>> damageCondition, Consumer<Pair<Entity, Entity>> bientityAction, Predicate<Pair<Entity, Entity>> bientityCondition) {
+        super(type, entity);
+        this.damageCondition = damageCondition;
+        this.bientityAction = bientityAction;
+        this.bientityCondition = bientityCondition;
+    }
+
+    public void onDeath(Entity target, DamageSource damageSource, float damageAmount) {
```

Comment:
I think it would make more sense if the context of actor and target is swapped for this power type, making the actor be the attacker and the target be the entity that was killed, which is the case with other power types (like `action_on_hit`/`action_when_hit`, `modify_damage_taken`/`modify_damage_dealt`, etc.)